### PR TITLE
Update Reviewers for PyTorch Distributed team

### PR DESCRIPTION
**Pull request by kurman**
Update merge rule approver list under 'Distributed' section based on current PyTorch distributed team composition.

### CODE REVIEW

**Review by fduwjj — APPROVED** [review]

File: .github/merge_rules.yaml [hunk 1]
```diff
@@ -275,14 +275,16 @@
...
   - wz337
   - awgu
   - fegin
-  - kumpera
-  - yhcharles
+  - kurman
+  - LucasLLC
+  - sanketpurandare
+  - shuqiangzhang
+  - tianyu-l
```
Comment (weifengpy) [inline .github/merge_rules.yaml:287]: - weifengpy

Hi Kurman, this is Wei Feng on FSDP. could I join the list?

**Review by XilunWu — APPROVED** [review]
lgtm! p.s. this PR may need a rebase since we already have `shuqiangzhang` in the newest version of the file.

File: .github/merge_rules.yaml (hunk 1 above)
Comment (XilunWu) [reply .github/merge_rules.yaml:287]: @weifengpy let's add your name in another PR: https://github.com/pytorch/pytorch/pull/117409
@kurman you can merge the PR if there's no conflict in rebasing. I added Wei's name in the above PR.

**Pull request closed without merging**